src/transform.ts

```diff
@@ -292,217 +354,201 @@ export function getIndexerTransactionFromAlgodTransaction(
+                        participant: new algosdk.indexerModels.StateProofParticipant({
+                          weight: Number(reveal.participant.weight),
+                          verifier: new algosdk.indexerModels.StateProofVerifier({
+                            keyLifetime: BigInt(reveal.participant.pk.keyLifetime),
+                            commitment: reveal.participant.pk.commitment,
+                          }),
+                        }),
+                      })
+                    })
+                  : undefined,
+              }),
+              message: new algosdk.indexerModels.IndexerStateProofMessage({
+                blockHeadersCommitment: Buffer.from(transaction.stateProof!.message!.blockHeadersCommitment),
+                firstAttestedRound: BigInt(transaction.stateProof!.message!.firstAttestedRound),
+                latestAttestedRound: BigInt(transaction.stateProof!.message!.lastAttestedRound),
+                lnProvenWeight: BigInt(transaction.stateProof!.message!.lnProvenWeight),
+                votersCommitment: Buffer.from(transaction.stateProof!.message!.votersCommitment),
+              }),
+              stateProofType: Number(transaction.stateProof!.stateProofType ?? 0),
+            }),
           }
         : undefined),
-      'first-valid': transaction.firstRound,
-      'last-valid': transaction.lastRound,
-      'tx-type': transaction.type,
+      firstValid: transaction.firstValid,
+      lastValid: transaction.lastValid,
+      txType: transaction.type,
       fee: transaction.fee ?? 0,
-      sender: algosdk.encodeAddress(transaction.from.publicKey),
-      'confirmed-round': roundNumber,
-      'round-time': roundTimestamp,
-      'intra-round-offset': roundOffset,
-      'created-asset-index': createdAssetId,
-      'genesis-hash': Buffer.from(transaction.genesisHash).toString('base64'),
-      'genesis-id': transaction.genesisID,
-      group: transaction.group ? Buffer.from(transaction.group).toString('base64') : undefined,
-      note: transaction.note ? Buffer.from(transaction.note).toString('base64') : undefined,
-      lease: transaction.lease ? Buffer.from(transaction.lease).toString('base64') : undefined,
-      'rekey-to': transaction.reKeyTo ? algosdk.encodeAddress(transaction.reKeyTo.publicKey) : undefined,
-      'closing-amount': closeAmount,
-      'created-application-index': createdAppId,
-      'auth-addr': blockTransaction.sgnr ? algosdk.encodeAddress(blockTransaction.sgnr) : undefined,
-      'inner-txns': blockTransaction.dt?.itx?.map((ibt) =>
+      sender: transaction.sender.toString(),
+      confirmedRound: BigInt(roundNumber),
+      roundTime: roundTimestamp,
+      intraRoundOffset: roundOffset,
+      createdAssetIndex: createdAssetId !== undefined ? BigInt(createdAssetId) : undefined,
+      genesisHash: transaction.genesisHash,
+      genesisId: transaction.genesisID,
+      group: transaction.group,
+      note: transaction.note,
```

Comment:
Previously this was a base64 string, do we not need to do that anymore?

Reply:
yes, `algosdk` updated this type to `Uint8Array`